test/cppsuite/tests/search_near_02.cxx

```diff
@@ -0,0 +1,430 @@
+                testutil_assert(k == key_prefix_str);
+            } else {
+                /* Both cursors should be pointing at the same key. */
+                testutil_assert(key_default_str == key_prefix_str);
+            }
+        }
+    }
+
+    /*
+     * Validate that no keys with the prefix used for the search have been found.
+     * To validate this, we can use the exact value set by the default search near. Since the prefix
+     * search near failed, the exact value set by the default search near call has to be either -1
+     * or 1:
+     * - If it is -1, we need to check the next key, if it exists, is lexicographically greater than
+     * the prefix we looked for.
+     * - If it is 1, we need to check the previous keys, if it exists, if lexicographically smaller
+     * than the prefix we looked for.
+     */
+    void
+    validate_unsuccessful_prefix_call(
+      scoped_cursor &cursor_default, const std::string &prefix, int exact_default)
+    {
+        int ret;
+        const char *k;
+        std::string k_str;
+
+        /*
+         * The exact value from the default search near call cannot be 0, otherwise the prefix
+         * search near should be successful too.
+         */
+        testutil_assert(exact_default == -1 || exact_default == 1);
+
+        /* Retrieve the key at the default cursor. */
+        const char *key_default;
+        testutil_check(cursor_default->get_key(cursor_default.get(), &key_default));
+        std::string key_default_str = key_default;
+
+        /* The key at the default cursor should not contain the prefix. */
+        testutil_assert(key_default_str.substr(0, prefix.size()) != prefix);
+
+        /* Example: */
+        /* keys: a, bb, bbb */
+        /* Default search_near(bba) returns bb, exact = -1 */
+        /* Prefix search_near(bba) returns WT_NOTFOUND */
+        if (exact_default == -1) {
+            /*
+             * The current key of the default cursor should be lexicographically smaller than the
+             * prefix.
+             */
+            testutil_assert(std::lexicographical_compare(
+              key_default_str.begin(), key_default_str.end(), prefix.begin(), prefix.end()));
+
+            /*
+             * The next key of the default cursor should be lexicographically greater than the
+             * prefix if it exists.
+             */
+            ret = cursor_default->next(cursor_default.get());
+            if (ret == 0) {
+                testutil_check(cursor_default->get_key(cursor_default.get(), &k));
+                k_str = k;
+                testutil_assert(!std::lexicographical_compare(
+                  k_str.begin(), k_str.end(), prefix.begin(), prefix.end()));
+            } else {
+                /* End of the table. */
+                testutil_assert(ret == WT_NOTFOUND);
+            }
+        }
+        /* Example: */
+        /* keys: a, bb, bbb */
+        /* Default search_near(bba) returns bba, exact = 1 */
```

Comment:
This here should return back "bbb", but logic looks good to me otherwie.

Reply:
Thanks, will fix the typo.

Reply:
Fixed in a5602ced362d3cf2d7bf95b2e4b8469689f4265a